discovery-provider/src/tasks/users.py

```diff
@@ -415,6 +424,47 @@ def update_user_associated_wallets(
         )
 
 
+class UserEventsMetadata(TypedDict):
+    referrer: int
+    is_mobile_user: bool
+
+
+def update_user_events(
+    session: Session,
+    user_record: User,
+    events: UserEventsMetadata
+) -> None:
+    """Updates the user events table"""
+    try:
+        if not isinstance(events, dict):
+            # There is something wrong with events, don't process it
+            return
+
+        # Mark existing UserEvents entries as not current
+        session.query(UserEvents).filter_by(user_id=user_record.user_id).update(
```

Comment:
might be faster if it also has `is_current=True` in the filter - not 100% sure, but index is on (user_id, is_current). can leave as is, should be fast regardless - userEvents table should be relatively small.

Reply:
it could be huge someday. good callout